nixos/modules/services/hardware/undervolt.nix

```diff
@@ -161,11 +173,16 @@ in
   };
 
   config = mkIf cfg.enable {
+    assertions = [ {
+      assertion = !hasCliArgs;
+      message = "At least one undervolt parameter must be set to enable service.undervolt";
+    } ];
+
     hardware.cpu.x86.msr.enable = true;
 
     environment.systemPackages = [ cfg.package ];
 
-    systemd.services.undervolt = {
+    systemd.services.undervolt = mkIf hasCliArgs {
```

Comment:
nit: after an assertion is added, there is not need to check `mkIf hasCliArgs` again.